src/cmap/connection_pool.ts

```diff
@@ -322,6 +324,15 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
     return this[kCheckedOut];
   }
 
+  get client() {
+    // NOTE: This is here to get tests to pass when using MockServer. Looking into a way to modify the testing
+    // infrastructure so it is unnecessary.
+    if (this[kServer].s?.topology) {
+      return this[kServer].s.topology.client;
+    }
```

Comment:
Note that this allows for testing to pass. Looking for where exactly it is failing so I can fix it there rather than in the production code